lib/ecto/registry.ex

```diff
@@ -36,8 +36,11 @@ defmodule Ecto.Registry do
   end
 
   def handle_info({:DOWN, ref, _type, pid, _reason}, table) do
-    [{^pid, ^ref, _}] = :ets.lookup(table, pid)
-    :ets.delete(table, pid)
+    case :ets.lookup(table, pid) do
+      [{^pid, ^ref, _}] -> :ets.delete(table, pid)
+      _ -> true
```

Comment:
Right, guess we can simply use `:ets.delete(table, pid)` then?

Reply:
I prefer the current code because it checks the `ref` so its explicit that this monitor is for this item in the table. Generally I think that ignoring the ref from the down is a code smell but it would be simpler so I won't approve or close this :).

Reply:
We could possibly use 
```elixir
1 = :ets.select_delete(table, [{{pid, ref, :_}, [], [true]}])
```
This will do just one ets operation and assert at the same time.